cdap-unit-test/src/main/java/co/cask/cdap/test/MetricsManager.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.test;
+
+import co.cask.cdap.api.dataset.lib.cube.AggregationFunction;
+import co.cask.cdap.api.dataset.lib.cube.TimeValue;
+import co.cask.cdap.api.metrics.MetricDataQuery;
+import co.cask.cdap.api.metrics.MetricStore;
+import co.cask.cdap.api.metrics.MetricTimeSeries;
+import co.cask.cdap.common.conf.Constants;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Used to query metrics with a context and metric-name.
+ */
+public class MetricsManager {
+
+  private MetricStore metricStore;
+
+  public MetricsManager(MetricStore metricStore) {
+    this.metricStore = metricStore;
+  }
+
+  public long getTotalCounter(Map<String, String> context, String metricName) {
+    MetricDataQuery query = getTotalCounterQuery(context, metricName);
+    return getSingleValueFromTotals(query);
+  }
+
+  public void resetAll() throws Exception {
+    metricStore.deleteAll();
+  }
+
+  public Collection<MetricTimeSeries> query(MetricDataQuery query) throws Exception {
+    return metricStore.query(query);
+  }
+
+
+  /**
+   * waitForFlowlet's events processed metric, specified by count, and the timeout and timOutUnit is provided.
+   * @param namespace
```

Comment:
Its not useful to have params in javadoc, if they're not going to have description, right?